argoverse/evaluation/stereo/eval.py

```diff
@@ -0,0 +1,208 @@
+
+        Returns:
+            Evaluation metrics.
+        """
+        pred_fpaths = list(self.pred_root_fpath.glob("**/*.png"))
+        gt_fpaths = list(self.gt_root_fpath.glob("**/stereo_front_left_rect_disparity/*.png"))
+        gt_obj_fpaths = list(self.gt_root_fpath.glob("**/stereo_front_left_rect_objects_disparity/*.png"))
+
+        if len(pred_fpaths) == 1:
+            timestamp = str(pred_fpaths[0]).split("_")[-1][:-4]
+            gt_fpaths = [gt_fpath for gt_fpath in gt_fpaths if timestamp in str(gt_fpath)]
+            gt_obj_fpaths = [gt_obj_fpath for gt_obj_fpath in gt_obj_fpaths if timestamp in str(gt_obj_fpath)]
+
+        assert len(pred_fpaths) == len(gt_fpaths)
+
+        if self.num_procs == 1:
+            errors = []
+            for (pred_fpath, gt_fpath, gt_obj_fpath) in zip(pred_fpaths, gt_fpaths, gt_obj_fpaths):
+                error = compute_disparity_error(
+                    pred_fpath,
+                    gt_fpath,
+                    gt_obj_fpath,
+                    self.figs_fpath,
+                    self.abs_error_thresholds,
+                    self.rel_error_thresholds,
+                    save_disparity_error_image=self.save_disparity_error_image,
+                )
+                errors.append(error)
+        else:
+            args = [
+                (
+                    pred_fpath,
+                    gt_fpath,
+                    gt_obj_fpath,
+                    self.figs_fpath,
+                    self.abs_error_thresholds,
+                    self.rel_error_thresholds,
+                    self.save_disparity_error_image,
+                )
+                for (pred_fpath, gt_fpath, gt_obj_fpath) in zip(pred_fpaths, gt_fpaths, gt_obj_fpaths)
+            ]
+            with Pool(self.num_procs) as p:
+                errors = p.starmap(compute_disparity_error, args)
+
+        data = pd.concat(errors)
+        # Sums over the instances along the rows containing the disparity error metrics.
+        data_sum = data.sum(axis=0)
+        summary: Dict[str, float] = {}
+
+        for abs_error_thresh in self.abs_error_thresholds:
+            d1_all = (data_sum[f"num_errors_bg:{abs_error_thresh}"] + data_sum[f"num_errors_fg:{abs_error_thresh}"]) / (
+                data_sum["num_pixels_bg"] + data_sum["num_pixels_fg"]
+            )
+            d1_fg = data_sum[f"num_errors_fg:{abs_error_thresh}"] / data_sum["num_pixels_fg"]
+            d1_bg = data_sum[f"num_errors_bg:{abs_error_thresh}"] / data_sum["num_pixels_bg"]
+            d1_all_est = (
+                data_sum[f"num_errors_bg_est:{abs_error_thresh}"] + data_sum[f"num_errors_fg_est:{abs_error_thresh}"]
+            ) / (data_sum["num_pixels_bg_est"] + data_sum["num_pixels_fg_est"])
+            d1_fg_est = data_sum[f"num_errors_fg_est:{abs_error_thresh}"] / data_sum["num_pixels_fg_est"]
+            d1_bg_est = data_sum[f"num_errors_bg_est:{abs_error_thresh}"] / data_sum["num_pixels_bg_est"]
```

Comment:
what does `d1` mean here?

Reply:
it was our old notation, following kitti. I'll remove those.